tests/src/JIT/HardwareIntrinsics/X86/Sse/StaticCast.cs

```diff
@@ -39,12 +45,41 @@ static unsafe int Main(string[] args)
                         Console.WriteLine();
                         testResult = Fail;
                     }
-                }
-            }
 
+                    try
```

Comment:
This code may be clear to others, but I had to read it twice before understanding how it is supposed to work (i.e. the successful path is the one that catches the exception, so that does nothing - it is the fall-through path that's the error path. I think comments here, as well as on the declaration of `Num` (i.e. that it is intentionally designed to be a `struct` type that meets the generic constraint but is not supported by any intrinsics) might make it easier to understand.

Reply:
Thanks, will fix.